library/core/src/alloc/global.rs

```diff
@@ -53,6 +53,25 @@ use crate::ptr;
 /// * `Layout` queries and calculations in general must be correct. Callers of
 ///   this trait are allowed to rely on the contracts defined on each method,
 ///   and implementors must ensure such contracts remain true.
+///
+/// * You may not rely on allocations actually happening, even if there are explicit
+///   heap allocations in the source. The optimizer may detect allocation/deallocation
+///   pairs that it can instead move to stack allocations/deallocations and thus never
+///   invoke the allocator here.
+///   More concretely, the following code example is unsound, irrespective of whether your
+///   custom allocator allows counting how many allocations have happened.
+///
+///   ```text
```

Comment:
```suggestion
///   ```rust,ignore (unsound)
```

Reply:
I had that, tidy complained

Reply:
It's not just unsound, it contains placeholders like `...` which obviously won't compile

Reply:
Tidy complained because you didn't mention a reason. The ` (unsound)` lets tidy pass (At least locally it worked, when a reason was provided).

Reply:
aah, thanks!